### ISSUE TITLE
Please add more detailed Azure deployment instructions including the Chat/dist path to select in VS Code

### ISSUE THREAD

**Issue by gbowerman** [Azure-Samples/communication-services-web-chat-hero#45]
Commands in package.json like:
"start": "cd Chat && npm run start && cd ..",
will fail when you deploy it to Azure App Service with:
 Invalid start-up command "cd Chat && npm run start && cd .." in package.json. Please use the format "node <script relative path>".

It works fine locally.

If someone has successfully deployed this example to Azure App Service, please share the exact instructions. Thanks!

### Minimal steps to reproduce
Deploy this to Azure App Service.

### Any log messages given by the failure
 Invalid start-up command "cd Chat && npm run start && cd .." in package.json. Please use the format "node <script relative path>".

### OS and Version?
Windows 10

**Comment by dmceachernmsft** [comment]
Hello! Thank you for filing the issue. When you attempted to deploy to your app service did you try with button in the README? Or the CLI route listed in the README? We have a known issue related to the button deployment method where you have to add the `endpointUrl` param required by chat to your `appsettings.json` file manually. 

I Will also look into if its a build issue when deploying using the Azure extension and get back to you shortly.

**Comment by gbowerman** [comment]
Thanks for the quick response. I was trying to follow this instruction after successfully testing locally: "Use the Azure extension and deploy the Chat/dist directory to your app service"
I installed the App Service VS Code extension and deployed using that. The first issue I encountered was that the App Service free tier runs out of space. After moving to a higher tier, the web.config still doesn't get generated when deploying to Azure. Based on the log file it appears to be the "Invalid start-up command" described above, which relates to the package.json files.

**Comment by dmceachernmsft** [comment]
Excellent thank you for the information. I will work through that flow and let you know what I find!

**Comment by dmceachernmsft** [comment]
Okay so I was able to deploy it just fine through the extension. Just so you know what I did the steps that I worked through were as follows:
- Add my `endpointUrl` and `resourceConnectionString` to my `appsettings.json` file.
- I then from the root directory of the repo ran:
  - `npm run setup`
  - `npm run build`
  - `npm run package`
- I then went to my azure app services extension signed in through the command palette `ctrl + shift + p` 
- after that I clicked on the following button and selected the app service I wanted to deploy to
![image](https://user-images.githubusercontent.com/94866715/163494141-53b4b64b-cbc4-4cb2-b490-c57b7dd6b0c5.png)
 - you should then get this drop down here 
![image](https://user-images.githubusercontent.com/94866715/163494269-8bc42a06-97f7-4c5e-9a1e-f17ceb2f67ba.png)
- you will want to browse only to this path here 
![image](https://user-images.githubusercontent.com/94866715/163494336-8a89ee9f-7f02-4a23-b356-03b3c5306540.png)
- if you choose to start a app service through the extension, or a existing one I spun up my new app service with node 14 LTS 

Let me know if this helps! My first thought though if you followed those same steps is that its possibly the version of node that you are running in your app service. If you ran in with 12 (which should work) try it with 14.

**Comment by gbowerman** [comment]
Thanks! I don't think I browsed to the /Chat/dist/ part of the path (I just picked the parent folder). That would be valuable info to include if this doc is updated.
I picked the latest node version in the list since I had the current version installed on my dev machine.
I will try again and make sure I pick the /Chat/dist path, and update here.

**Comment by gbowerman** [comment]
Update. Thank you @dmceachernmsft my app is working now.
If it's ok with you I'll edit the subject of this issue, to include the path to browse to (communication-services-web-chat/Chat/dist/) when deploying. The information you provided here solved my problem, and I'd love to see it in the readme.

**Comment by dmceachernmsft** [comment]
Thank you for the feedback I will be sure to raise this with the team next week and we'll get something in there to drive more clarity on deployment!

Have a great weekend!

**Comment by gbowerman** [comment]
@dmceachernmsft one concern, while the first page works, the app hangs on "Creating a new chat thread..." (it worked correctly running locally). Was there an equivalent Server/dist that needed to be deployed as well?

**Comment by dmceachernmsft** [comment]
When you built the progect did you have your ACS `endpointUrl` and `ResourceConnectionString` set in the `appsettings.json` file? those strings should be there when you build the project. you can find them in this menu option in your ACS resource menu in Azure Portal
![image](https://user-images.githubusercontent.com/94866715/163625038-9fc17541-ee32-4cae-a9fc-5b9445d0cae2.png)

You can also add them in the configuration sub menu in your app service, you can find it in the menu on the left when viewing your app service in Azure portal. 
![image](https://user-images.githubusercontent.com/94866715/163625097-30fe44b6-f7ce-42db-97d4-ec4aca866162.png)
you will need to add the endpointUrl manually on this screen if you dont add the strings in the source code.
![image](https://user-images.githubusercontent.com/94866715/163625246-bac810fd-56f7-4d33-8a65-83717c265177.png)

I hope this helps!

**Comment by gbowerman** [comment]
> When you built the project did you have your ACS `endpointUrl` and `ResourceConnectionString` set in the `appsettings.json` file? 
Yes. The app works fine running on localhost:3000, which implies the appsettings.json values are correct. The hang on creating a new chat thread only happens in App Service.

**Comment by dmceachernmsft** [comment]
you can check in the built server.js file you can search for those values. If they are missing for whatever reason try adding them through the Azure portal.

Hope this helps!

**Comment by gbowerman** [comment]
The correct values are in server.js on the build machine, but looking in App Service using the Console, there's just a build.js and index.html in the c:\home\site\wwwroot folder. How does the server.js get to the App Service machine if we select communication-services-web-chat/Chat/dist/build as the source when deploying?

**Comment by dmceachernmsft** [comment]
So this is what the www/root folder should look like: 
![image](https://user-images.githubusercontent.com/94866715/163834696-0de1351f-b2eb-459c-8773-32a448fc84e9.png)
in your case the `build` and `dist-esm` folder dosen't exist? its just the `build.js` and `index.html` file?

if you are able to run the app and the app loads could you send along some screen shots of the errors your are seeing with dev tools in the console, or in the log stream in your app service? You can turn on the logs here in App service logs, and see whats happening in log stream
![image](https://user-images.githubusercontent.com/94866715/163836581-c0b24061-74cb-4889-9fb6-5bf6f229ad81.png)
You will be able to see requests the app is making from the application stream and if there is any problems with the connections. 

Hope you had a great weekend and this helps!

**Comment by alkwa-msft** [comment]
![image](https://user-images.githubusercontent.com/79329532/163837714-2a2ae42a-f380-4309-836e-88ca0cfa14cc.png)

Did we run` npm run package`? It should copy the server.js into the dist directory before you deploy your app to the app service.

**Comment by gbowerman** [comment]
@alkwa-msft @dmceachernmsft - thank you! Confirming it's fully working in App Service now.
What I did wrong: misinterpret this screenshot:
![image](https://user-images.githubusercontent.com/4258913/163886214-2aaf7629-e8b1-4a67-b35d-8a96409a5cd7.png)
"build" is highlighted so I thought I was supposed to pick a path of Chat/dist/build - now I've corrected this to Chat/dist it works correctly.
The instructions in the current version of the doc are correct. However detailed instructions/screenshots for using the VS Code App Service extension would be useful, as the single line starting "Use the Azure extension" with lower case "app service" is not going to be meaningful to someone unfamiliar with the VS Code App Service extension, or to someone unfamiliar with App Service (what service plan to create, etc..).
Thank you for sticking with me. Glad it's working now, and with hindsight, simple :-)
